src/main/protobuf/bazel_output_service.proto

```diff
@@ -0,0 +1,324 @@
+  // base.
+  //
+  // Starting a build finalizes any previous build with the same output_base_id
+  // that has not been finalized yet as if a FinalizeBuildRequest had been sent
+  // with build_successful set to false.
+  string output_base_id = 2;
+
+  // A client-chosen value that uniquely identifies this build. This value must
+  // be provided to most other methods to ensure that operations are targeted
+  // against the right output path.
+  //
+  // Bazel sets this value to --invocation_id.
+  string build_id = 3;
+
+  // Additional arguments to pass depending on how Bazel communicate with the
+  // Content Addressable Storage.
+  google.protobuf.Any args = 4;
+
+  // The absolute path at which the remote output service exposes its output
+  // paths, as seen from the perspective of the client.
+  //
+  // This value needs to be provided by the client, because file system
+  // namespace virtualization may cause this directory to appear at a location
+  // that differs from the one used by the service.
+  //
+  // The purpose of this field is to ensure that the remote output service is
+  // capable of expanding symbolic links containing absolute paths.
+  //
+  // If this is not set, or an empty string, the service must determine where to
+  // expose its output path and return an absolute path in
+  // StartBuildResponse.output_path_suffix.
+  string output_path_prefix = 5;
+
+  // A map of paths on the system that will become symbolic links pointing to
+  // locations inside the output path. Similar to output_path_prefix, this
+  // option is used to ensure the remote output service is capable of expanding
+  // symbolic links.
+  //
+  // Map keys are absolute paths, while map values are paths that are
+  // relative to the output path.
+  map<string, string> output_path_aliases = 6;
+}
+
+message StartBuildResponse {
+  // If set, the contents of the output path are almost entirely identical on
+  // the results of a previous build. This information may be used by Bazel to
+  // prevent unnecessary scanning of the file system.
+  //
+  // The server MUST leave this field unset in case the contents of the output
+  // path are empty, not based on a previous build, if no tracking of this
+  // information is performed, or if the number of changes made to the output
+  // path is too large to be expressed.
+  InitialOutputPathContents initial_output_path_contents = 1;
+
+  // A path that the client must append to StartBuildRequest.output_path_prefix
+  // to obtain the full path at which outputs of the build are stored.
+  //
+  // Bazel replaces bazel-out/ with a symlink targeting this path.
+  string output_path_suffix = 2;
+}
+
+message InitialOutputPathContents {
+  // The identifier of a previously finalized build whose results are stored in
+  // the output path.
+  string build_id = 1;
+
+  // Output files that have been modified (either the content or metadata) or
+  // removed since the build was finalized.
```

Comment:
This case is documented at `StartBuildResponse.initial_output_path_contents`:

```
  // The server MUST leave this field unset in case the contents of the output
  // path are empty, not based on a previous build, if no tracking of this
  // information is performed, or if the number of changes made to the output
  // path is too large to be expressed.
```